stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -995,6 +997,14 @@ public static String getSQLType(BType value) {
             return Constants.SQLDataTypes.BOOLEAN;
         case TypeTags.DECIMAL_TAG:
             return Constants.SQLDataTypes.DECIMAL;
+        case TypeTags.ARRAY_TAG:
+            if (value instanceof BArrayType) {
+                if (((BArrayType) value).getElementType().getTag() == TypeTags.BYTE_TAG) {
```

Comment:
Can't we merge the two if conditions?

Also I think the exception should be thrown in the else block of that if condition. This error should be more specific such that it indicates that the problem is passing an array of a type other than byte.

Also, can we add a negative test to verify this exception is thrown appropriately?

Reply:
I have tried to add negative test to check the exception is thrown by adding int[] into database as direct parameter.

but it gives compilation error as,
```
incompatible types: expected 'string|int|boolean|float|decimal|byte[]|ballerina/sql:Parameter', found 'int[]'
```